File: Longest Consecutive Sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums)==0:
            return 0
        num_set=set(nums)
        count = 1
        mx_count=1
        y = float('-inf')
        for x in sorted(num_set):
            if x==y+1:
                count+=1
                mx_count=max(mx_count, count)
                y=x
            else:
                count = 1
                y=x
        return mx_count

File: Longest Consecutive Sequence.cpp
class Solution {
    public:
        int longestConsecutive(vector<int>& nums) {
            if(nums.size()==0)
              return 0;
            set<int>s;
            for(auto num: nums)
            {
                s.insert(num);
            }
            int count=1, y=INT_MIN, mx_count=1;
            for(auto x: s)
            {
                if(x==y+1)
                {
                    count++;
                    mx_count=max(mx_count, count);
                    y=x;
                }
                else
                {
                    count=1;
                    y=x;
                }
            }
            return mx_count;
        }
    };